src/v0/destinations/sfmc/transform.js

```diff
@@ -271,11 +283,41 @@ const responseBuilderSimple = async ({ message, destination, metadata }, categor
   throw new ConfigurationError(`Event type '${category.type}' not supported`);
 };
 
+const retlResponseBuilder = async (message, destination, metadata) => {
+  const { clientId, clientSecret, subDomain, externalKey } = destination.Config;
+  const token = await accessTokenCache.get(metadata.destinationId, () =>
+    getToken(clientId, clientSecret, subDomain, metadata),
+  );
+  const { destinationExternalId, objectType, identifierType } = getDestinationExternalIDInfoForRetl(
+    message,
+    'SFMC',
+  );
+  if (objectType.toLowerCase() === 'data extension') {
```

Comment:
We can have a check if somehow objectType comes null or undefined

Reply:
addressed